### PR TITLE
fix(adk): fix ToolCall Arguments format in GenTransferMessages

### PR DESCRIPTION
**Pull request by shenysun**
#### What type of PR is this?
fix

#### Check the PR title.
- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.
修复 GenTransferMessages 中的 ToolCall Arguments 格式问题

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

This PR fixes the ToolCall Arguments format in the `GenTransferMessages` function. Previously, the Arguments field directly used the destination agent name as a plain string, which doesn't conform to the standard JSON object format for tool calls. This would cause tool call parsing to fail in supervisor scenarios when transferring between agents.

The fix changes the Arguments from:
```
Arguments: destAgentName
```

To the proper JSON object format:
```
Arguments: '{"agent_name": "destAgentName"}'
```

**Files Changed:**
- `adk/utils.go`: Added JSON marshaling for tool call arguments
- `adk/utils_test.go`: Added comprehensive unit tests for GenTransferMessages

**Testing:**
- Added TestGenTransferMessages: Verify basic functionality including tool call structure, JSON argument format, and message content
- Added TestGenTransferMessages_EmptyAgentName: Test edge case with empty agent name
- Added TestGenTransferMessages_SpecialCharactersInAgentName: Validate handling of special characters in agent names
- Run supervisor transfer scenarios to verify correct parameter parsing
- All tests pass successfully ✅

### CODE REVIEW

File: adk/utils.go [hunk 1]
```diff
@@ -87,7 +88,9 @@ func concatInstructions(instructions ...string) string {
 // transfer-to-agent tool call targeting the destination agent.
 func GenTransferMessages(_ context.Context, destAgentName string) (Message, Message) {
 	toolCallID := uuid.NewString()
-	tooCall := schema.ToolCall{ID: toolCallID, Function: schema.FunctionCall{Name: TransferToAgentToolName, Arguments: destAgentName}}
+	args := map[string]string{"agent_name": destAgentName}
+	argsJSON, _ := json.Marshal(args)
+	tooCall := schema.ToolCall{ID: toolCallID, Function: schema.FunctionCall{Name: TransferToAgentToolName, Arguments: string(argsJSON)}}
```
Comment (shentongmartin) [inline adk/utils.go:94]: 具体啥情况下会需要这个？

Comment (shenysun) [reply adk/utils.go:94]: 1. 我们的 LLM API 基于 OpenAI 协议封装，对 tool_calls 字段有严格校验。多轮对话中传递历史消息时，若格式不符合规范会报错。
2. 在 OpenAI 协议中，tool_calls arguments 定义为 JSON-encoded 格式。
https://platform.openai.com/docs/guides/function-calling?api-mode=chat#handling-function-calls

image

Comment (shentongmartin) [reply adk/utils.go:94]: openai 模型并不会因为这个 argument 不是 json 而报错

Comment (shenysun) [reply adk/utils.go:94]: 确实，OpenAI 模型本身不会因为这个报错。这个问题是我们内部 LLM API 封装层的校验逻辑导致的，不是框架的 bug。

不过从协议规范角度，tool_calls.arguments 确实应该是 JSON-encoded 格式。我的修改让代码更符合 OpenAI 官方规范。

Comment (shentongmartin) [reply adk/utils.go:94]: "JSON-encoded arguments"，这个的字面意思不是一定是 object。简单的一个 string 也是 json-encoded.

Comment (shenysun) [reply adk/utils.go:94]: 您说的没毛病，从 JSON 规范角度 `"hello"` 确实是 JSON-encoded。

但我的修改考虑的是：

1. 工具定义的对应关系：`transfer_to_agent` 工具在代码中明确定义了参数是对象类型

```go
// adk/chatmodel.go:417-427
toolInfoTransferToAgent = &schema.ToolInfo{
    Name: TransferToAgentToolName,
    ParamsOneOf: schema.NewParamsOneOfByParams(map[string]*schema.ParameterInfo{
        "agent_name": {
            Desc:     "the name of the agent to transfer to",
            Required: true,
            Type:     schema.String,
        },
    }),
}
```

工具的参数定义是一个包含 `agent_name` 字段的对象，而不是简单字符串。

2. 解析代码的期望（目前这个场景不会解析，deterministic_transfer 直接构造的tool msg）：`transfer_to_agent` 的实现使用 `UnmarshalString` 解析参数

```go
// adk/chatmodel.go:495-500
type transferParams struct {
    AgentName string `json:"agent_name"`
}

params := &transferParams{}
err := sonic.UnmarshalString(argumentsInJSON, params)
```

这段代码期望 `argumentsInJSON` 是一个 JSON 对象字符串 `{"agent_name": "xxx"}`，才能正确解析到 `params.AgentName`。
